controllers/volumereplicationgroup_controller.go

```diff
@@ -462,27 +468,132 @@ func (v *VRGInstance) createVolumeReplicationResources() bool {
 
 		volRep := &volrep.VolumeReplication{}
 
-		err := v.reconciler.Get(v.ctx, types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, volRep)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				log.Info("Creating VolumeReplication resource for PersistentVolumeClaim")
+		err := v.reconciler.Get(v.ctx, pvcNamespacedName, volRep)
+		if err != nil && errors.IsNotFound(err) {
+			log.Info("Uploading PersistentVolume")
 
-				if err = v.createVolumeReplicationForPVC(pvc); err == nil {
-					continue
-				}
+			if err = v.uploadPV(pvc); err != nil {
```

Comment:
pvc is passed as an argument to uploadPV here instead of PV.

Reply:
I probably commented bit early :). I see that in uploadPV function, the PersistentVolume of the pvc is obtained and used.

Reply:
FYI: an earlier version of the method name was uploadPVofPVC(), which was later changed to a simpler uploadPV().